quesma/v2/core/quesma_logger.go

```diff
@@ -121,6 +132,45 @@ func (l *QuesmaLoggerImpl) ErrorWithCtx(ctx context.Context) *zerolog.Event {
 	return event
 }
 
+func (l *QuesmaLoggerImpl) DeduplicatedInfo() DeduplicatedEvent {
+	return DeduplicatedEvent{
+		event: l.Info(),
+		l:     l,
+	}
+}
+
+func (l *QuesmaLoggerImpl) DeduplicatedWarn() DeduplicatedEvent {
+	return DeduplicatedEvent{
+		event: l.Warn(),
+		l:     l,
+	}
+}
+
+type DeduplicatedEvent struct {
+	event *zerolog.Event
+	l     *QuesmaLoggerImpl
+}
+
+func hashMsgf(format string, v ...interface{}) uint32 {
+	// []interface{} is not hashable, so we need to hash it manually
+	// For the convenience sake we just hash a Print representation
+	h := fnv.New32a()
+	fmt.Fprint(h, format, v)
+	return h.Sum32()
+}
+
+// TODO: Add wrappers for other *zerolog.Event methods
+func (m DeduplicatedEvent) Msgf(format string, v ...interface{}) {
+	hash := hashMsgf(format, v)
```

Comment:
Non blocking, can we somehow test deduplication logic?

Reply:
Added a unit test: https://github.com/QuesmaOrg/quesma/pull/1266/commits/197bb87c5789cd359aa65a67c6c45a281f2e9ff4